src/module/reservation.js

```diff
@@ -0,0 +1,142 @@
+  const url = `https://www.themealdb.com/api/json/v1/1/lookup.php?i=${idMeal}`;
+  const getData = async () => {
+    const response = await fetch(url);
+    const data = await response.json();
+    return data;
+  };
+  const lookup = getData();
+  lookup.then((data) => {
+    const mealName = data.meals[0].strMeal;
+    const mealImage = data.meals[0].strMealThumb;
+    const tags = data.meals[0].strTags;
+    popup.classList.add('popup');
+    popup.innerHTML = `
+    <img class = "popup-Images" src="${mealImage}" alt="Food Image">
+    <div class="popup-inner" id="start">
+        <article class="popup-header">
+            <button id="end"><svg class ="endALl" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><!--! Font Awesome Pro 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2022 Fonticons, Inc. --><path d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"/></svg></button>
+        </article>
+        <article class="popup-body">
+            <h2 class ="mealName">${mealName}</h2>
+            <div class="card-tags">
+                <ul class = "popDet">
+                <li>Country : ${data.meals[0].strArea}</li>
+                <li>Catogory : ${data.meals[0].strCategory}</li>
+                </ul>
+                <ul class = "popDet2">
+                <li class = "popup-Tags"> MealTags : ${tags}</li>
+                </ul>
+            </div>
+       </article>
+       <article class = "popup-footer">
+            <form class="form-inline" id = "makeRes">
+                 <h3 id ="headForm"> Make a Reservation </h3>
+                <input type="name" class="form-control" placeholder="Your Name" !required></input>
+                <label for="date">Start Date</label>
+                <input type="date" class="form-control" id="start-date" !required></input>
+                <label for="date">End Date</label>
+                <input type = "date" class="form-control" id="end-date" !required></input>
+                <button type = "submit" class="btn btn-primary" id="submitReservation">Reserve!</button>
+            </form>
+        </article>
+    `;
+    const currentReservations = document.createElement('ul');
+    currentReservations.classList.add('res');
+    const form = document.querySelector('.popup-footer');
+    const inner = document.querySelector('.popup-inner');
+    inner.insertBefore(currentReservations, form);
+    const url2 = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/FXMctN9lLmEpmOxXkx1x/reservations?item_id=${idMeal}`;
+    const getRes = async () => {
+      const res = await fetch(url2);
+      const info = await res.json();
+      return info;
+    };
```

Comment:
I think it will be better and clear to separate the function and the DOM. Please if you don't mind you can create a module to have all your calls to the API